plugins/out_s3/s3.c

```diff
@@ -1331,15 +1356,16 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
     }
     flb_sds_destroy(s3_key);
     uri = tmp;
-
-    if (ctx->compression != FLB_AWS_COMPRESS_NONE) {
+    if (ctx->compression != FLB_AWS_COMPRESS_NONE && ctx->is_compressed == FLB_FALSE) {
         ret = flb_aws_compression_compress(ctx->compression, body, body_size,
                                           &compressed_body, &final_body_size);
         if (ret == -1) {
             flb_plg_error(ctx->ins, "Failed to compress data");
             flb_sds_destroy(uri);
+            flb_free(compressed_body);
```

Comment:
I think this is not needed since if the compression function fails, the compressed_body will not be used?

Reply:
Not sure is there a chance, ret failed, compressed_body still have some data. But I can remove it